packages/core/src/auth/activation.ts

```diff
@@ -3,16 +3,32 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+import * as vscode from 'vscode'
 import { Auth } from './auth'
 import { LoginManager } from './deprecated/loginManager'
 import { fromString } from './providers/credentials'
 import { getLogger } from '../shared/logger'
 import { ExtensionUse } from './utils'
-import { isCloud9 } from '../shared/extensionUtilities'
+import { isAmazonQ, isCloud9, isSageMaker } from '../shared/extensionUtilities'
 import { isInDevEnv } from '../shared/vscode/env'
 import { isWeb } from '../shared/extensionGlobals'
+import { CredentialsProviderManager } from './providers/credentialsProviderManager'
+import { SharedCredentialsProviderFactory } from './providers/sharedCredentialsProviderFactory'
+import { Ec2CredentialsProvider } from './providers/ec2CredentialsProvider'
+import { EcsCredentialsProvider } from './providers/ecsCredentialsProvider'
+import { EnvVarsCredentialsProvider } from './providers/envVarsCredentialsProvider'
+
+interface SagemakerCookie {
+    authMode?: string
+}
 
 export async function initialize(loginManager: LoginManager): Promise<void> {
+    if (isAmazonQ() && isSageMaker()) {
+        const result = (await vscode.commands.executeCommand('sagemaker.parseCookies')) as SagemakerCookie
```

Comment:
Where does this command come from? Is it already registered in VSC by sagemaker? If so add a comment to mention this.

Reply:
Correct, this command is registered in Sagemaker code editor. I'll add a comment

Reply:
Lets make a new sagemaker folder in `src/shared/sagemaker` for all custom sagemaker code so that it is contained within a single place.

Reply:
This isn't a long-term approach. The Toolkit/Q extensions already have an api to accept connections. If needed, they can be enhanced to accept tokens/credentials, so the caller (SM in this case) can inject credentials into the extension.